src/pages/signup/index.jsx

```diff
@@ -1,7 +1,174 @@
-import React from 'react';
+import React, { useState } from 'react';
+import bcrypt from 'bcryptjs';
+import isError from '../../utils/signUpFuncs';
 
 const SignUp = () => {
-  return <div>Hello Signup</div>;
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [phone, setPhone] = useState('');
+  const [password1, setPassword1] = useState('');
+  const [password2, setPassword2] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showNoti, setShowNoti] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const data = {
+      email,
+      username,
+      phone,
+      password1,
+      password2,
+    };
+    const err = isError(data);
+    setErrors(err);
+
+    if (Object.keys(err).length === 0) {
+      const hash = bcrypt.hashSync(password1, 8);
+
+      const users = JSON.parse(localStorage.getItem('users'));
+      const newUser = {
+        id: users.length,
+        email,
+        username,
+        password: hash,
+        phone,
+      };
+
+      users.push(newUser);
+      localStorage.setItem('users', JSON.stringify(users));
+
+      setShowNoti(true);
+    } else setShowNoti(false);
+  };
+
+  return (
+    <div className="h-screen bg-gray-100 flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
+      <div className="max-w-md w-full space-y-8">
+        {showNoti ? (
+          <div
+            className="p-4 mb-4 text-sm text-green-700 bg-green-100 rounded-lg dark:bg-green-200 dark:text-green-800"
+            role="alert">
+            <span className="font-medium">Sign up sucessfully!</span>{' '}
+            Refresh localStorage to see new account.
+          </div>
+        ) : (
+          ''
+        )}
+        <div>
+          <img
+            className="mx-auto h-12 w-auto"
+            src="https://tailwindui.com/img/logos/workflow-mark-indigo-600.svg"
+            alt="Workflow"
+          />
+          <h2 className="mt-6 text-center text-3xl font-extrabold text-gray-900">
+            Create your new account
+          </h2>
+        </div>
+        <form className="mt-8 space-y-6" action="#" method="POST">
+          <div className="rounded-md shadow-sm -space-y-px">
+            <div>
+              <label htmlFor="email" className="sr-only">
+                email
+              </label>
+              <input
+                id="email"
+                name="email"
+                type="email"
+                autoComplete="email"
+                className="input-form"
+                placeholder="Email address"
+                required
+                value={email}
+                onChange={(e) => setEmail(e.target.value)}
+              />
```

Comment:
We can make the label, input into a common Input component because there are many logics and props are similar

Reply:
so we can remove the label tag and only use the input tag?

Reply:
Nah, I mean that put the label and input of a specific field into a separate component, just like this

<img width="596" alt="Screen Shot 2022-07-09 at 22 53 23" src="https://user-images.githubusercontent.com/107382512/178113122-da43b680-4041-4b95-bfca-65a51b5bcce5.png">

Reply:
Oh, understood